### PR TITLE
Expand test functions for Mesh Class

### PR DESCRIPTION
**Pull request by Licini**
1.expanded test functions to cover the majority of apis from Mesh class

2.Fixed few minor obvious bugs from _mesh.py. 

3.Applied OrderedDict in obj importer and one of mesh api to maintain key order accross py2/3 so they could pass few new testing functions.

### What type of change is this?

- [x] Bug fix in a **backwards-compatible** manner.
- [x] New feature in a **backwards-compatible** manner.
- [ ] Breaking change: bug fix or new feature that involve incompatible API changes.
(some of new tests will only pass with implementation of OrderedDict this time)
### Checklist

1. [x] Add the change to the `CHANGELOG.md` file in the `Unreleased` section under the most fitting heading: `Added`, `Changed`, `Removed`.
1. [x] Run all tests on your computer (i.e. `invoke test`).
1. [ ] If you add new functions/classes, check that:
   1. [ ] Are available on a second-level import, e.g. `compas.datastructures.Mesh`.
   1. [ ] Add unit tests (especially important for algorithm implementations).

### CODE REVIEW

File: src/compas/datastructures/mesh/_mesh.py [hunk 1]
```diff
@@ -1339,7 +1340,7 @@ def is_valid(self):
                 if self.halfedge[u][v] is None and self.halfedge[v][u] is None:
                     return False
                 fkey = self.halfedge[u][v]
-                if fkey:
+                if fkey or fkey == 0:
```
Comment (tomvanmele) [inline src/compas/datastructures/mesh/_mesh.py]: should this be `if fkey is not None`

**Review by gonzalocasas — COMMENTED** [review]
LGTM!

Only two questions: 
1. Do we want to leave the empty test functions (e.g. `def test_clear_halfedgedict`) as a reminder that we need to add them? or should we just get rid them them and only add when we really add the test?
2. You mention it's a breaking change, but I am not sure, I thought `OrderedDict` is mostly a drop-in replacement of `dict`, and then only thing that would break is if someone is doing explicit type-checking on this, which they shouldn't anyway.  Or is there's some other breaking behavior on the ordered dict?

File: src/compas/datastructures/mesh/_mesh.py [hunk 2]
```diff
@@ -2617,7 +2618,7 @@ def vertex_curvature(self, vkey):
 
         """
 
-        return 2 * pi - sum([angle_points(mesh.vertex_coordinates(vkey), mesh.vertex_coordinates(u), mesh.vertex_coordinates(v)) for u, v in pairwise(self.vertex_neighbors(vkey, ordered = True) + self.vertex_neighbors(vkey, ordered = True)[:1])])
+        return 2 * pi - sum([angle_points(self.vertex_coordinates(vkey), self.vertex_coordinates(u), self.vertex_coordinates(v)) for u, v in pairwise(self.vertex_neighbors(vkey, ordered = True) + self.vertex_neighbors(vkey, ordered = True)[:1])])
```
Comment (tomvanmele) [inline src/compas/datastructures/mesh/_mesh.py:2626]: we should split up this kind of unreadableness. i think i showed one too many times how great one-liners are and now some of our guys write everything they can that way :)

**Comment by Licini** [comment]
Thanks for review, @gonzalocasas, my thoughts on your questions :
1. I am more intended to leave them as reminders so later people don't have to browse through the source code to find out what to add. (I'm also thinking to make some generator that can create draft test files and just fill in actual testing code for each functions)

2. I guess I got a bit semiological confusion there. It is some of new tests added in will not pass without OrderedDict. After thinking a bit more, it seems to be backwards-compatible instead. Gonna make a change to the note

**Review by tomvanmele — APPROVED** [review]
just a few notes

File: src/compas/datastructures/mesh/_mesh.py (hunk 1 above)
Comment (Licini) [reply src/compas/datastructures/mesh/_mesh.py]: good point! changed

File: src/compas/datastructures/mesh/_mesh.py (hunk 2 above)
Comment (Licini) [reply src/compas/datastructures/mesh/_mesh.py:2626]: lol, this was an impressive one-liner!

**Comment by gonzalocasas** [comment]
@brgcode could we merge this? or any pending remarks?

**Pull request merged**